thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/HashService.js

```diff
@@ -112,10 +112,41 @@ HashService.prototype = {
     this.currentController = this.controllerNameToControllerMap[controllerName];
     this.refreshWindowHashForRouting(controllerName);
   },
-  onHashChangeEventHandler : function() {
-    console.log('OnHashChangeEventhandler');
-    console.log(this.params);
-    console.log(this.currentController);
-    this.currentController.handleAppEvent();
+
+  /**
+   * determine based on the hash change event if a transition is needed
+   * @param  {string} options.oldURL [oldURL of hash change event]
+   * @param  {string} options.newURL [newURL of hash change event]
+   * @return {boolean} - returns true if a transition is needed]
+   */
+  shouldTransition({ oldURL, newURL }) {
+    const [_1, transitionFrom, paramsFrom]= oldURL.split(/(?:[#?]|[&]*rand)/);
```

Comment:
Might be helpful to add a comment here on what the intention behind matching the charaters `rand` after the quantifier